java/gazelle/resolve.go

```diff
@@ -105,48 +103,34 @@ func (jr Resolver) Resolve(c *config.Config, ix *resolve.RuleIndex, rc *repo.Rem
 			continue
 		}
 
-		addDep(dep.String(), imp)
+		deps.Add(dep)
 	}
 
-	if len(deps) > 0 {
-		var labels []label.Label
-		for key := range deps {
-			l, err := label.Parse(key)
-			if err != nil {
-				jr.lang.logger.Fatal().Str("key", key).Err(err).Msg("cannot parse label")
-			}
-
+	if deps.Len() > 0 {
+		var exprs []build.Expr
+		for _, l := range deps.SortedSlice() {
 			if l.Relative && l.Name == from.Name {
 				continue
 			}
-
-			labels = append(labels, l)
-		}
-
-		sort.Slice(labels, func(i, j int) bool {
-			if labels[i].Relative {
-				if labels[j].Relative {
-					return labels[i].String() < labels[j].String()
-				}
-
-				return true
-			}
-
-			if labels[j].Relative {
-				return false
-			}
-
-			return labels[i].String() < labels[j].String()
-		})
-
-		var exprs []build.Expr
-		for _, l := range labels {
 			exprs = append(exprs, &build.StringExpr{Value: l.String()})
 		}
 		r.SetAttr("deps", exprs)
 	}
 }
 
+func labelLess(l, r label.Label) bool {
+	if l.Relative {
+		if r.Relative {
+			return l.String() < r.String()
+		}
+		return true
+	}
+	if r.Relative {
+		return false
+	}
+	return l.String() < r.String()
```

Comment:
Is there a way to get the fully-qualified label? If there is, then this would be enough of a check

Reply:
That's exactly what `.String()` does on a `Label` :)

Reply:
Then isn't this final check enough, so you don't need the additional checks above? That is, this entire method could be boiled down to this final line?

Reply:
Not quite - `/` sorts before `:` in UTF-8, so just sorting by stringification would put absolute labels before relative ones. Added a comment to the code explaining why we do this.